### PR TITLE
String#partition

### PR DESCRIPTION
**Pull request by johnjansen**

### CODE REVIEW

File: src/string.cr [hunk 1]
```diff
@@ -2200,6 +2200,45 @@ class String
...
+  # Searches sep or pattern (regexp) in the string,
+  # and returns the part before it, the match, and the part after it.
+  # If it is not found, returns two empty strings and str.
+  #
+  # ```
+  # "hello".rpartition("l") # => {"he", "l", "lo"}
+  # "hello".rpartition("x") # => {"hello", "", ""}
+  # ```
+  def partition(search : (Char | String)) : Tuple(String, String, String)
+    pre, mid, post = {"", "", ""}
```
Comment (asterite) [inline src/string.cr]: I think this is much simpler:

```cr
pre = mid = post = ""
```

But in fact maybe you could assign these variables in each case, or maybe just return a different tuple in every case branch.

Comment (johnjansen) [inline src/string.cr]: ooooh i like that .... cant believe i missed it ;-)

File: src/string.cr [hunk 2]
```diff
@@ -2200,6 +2200,45 @@ class String
     match_result.try &.begin(0)
   end
 
+  # Searches sep or pattern (regexp) in the string,
+  # and returns the part before it, the match, and the part after it.
+  # If it is not found, returns two empty strings and str.
```
Comment (johnjansen) [inline src/string.cr]: crap, thats what happens when you try to rush a feature through at midnight

File: src/string.cr (hunk 1 above)
Comment (asterite) [inline src/string.cr]: It's ok, in any case the performance should be the same. You can also do:

```cr
pre, mid, post = "", "", ""
```

That is, you don't need the tuple on the right. But the shortest one, maybe the clearest one, is the chained assignment.

**Comment by asterite** [comment]
Thank you! ❤️

**Pull request merged**